db/hybrid_logical_vector.go

```diff
@@ -122,8 +148,16 @@ func (hlv *HybridLogicalVector) IsInConflict(otherVector HybridLogicalVector) bo
 
 // AddVersion adds newVersion to the in memory representation of the HLV.
 func (hlv *HybridLogicalVector) AddVersion(newVersion Version) error {
-	if newVersion.Value < hlv.Version {
-		return fmt.Errorf("attempting to add new version vector entry with a CAS that is less than the current version CAS value. Current cas: %d new cas %d", hlv.Version, newVersion.Value)
+	var newVersionCAS uint64
+	if newVersion.Value == hlvExpandMacroCASValue {
+		// if we have macro expand value then we need to add max uint64 here for below comparison
+		newVersionCAS = math.MaxUint64
+	} else {
+		newVersionCAS = base.HexCasToUint64(newVersion.Value)
+	}
+	hlvVersionCAS := base.HexCasToUint64(hlv.Version)
+	if newVersionCAS < hlvVersionCAS {
+		return fmt.Errorf("attempting to add new version vector entry with a CAS that is less than the current version CAS value. Current cas: %s new cas %s", hlv.Version, newVersion.Value)
```

Comment:
Could possibly do this to prevent future accidental misuse of `newVersionCAS = math.MaxUint64`

```suggestion
	hlvVersionCAS := base.HexCasToUint64(hlv.Version)
	if newVersion.Value != hlvExpandMacroCASValue {
		newVersionCAS = base.HexCasToUint64(newVersion.Value)
		if newVersionCAS < hlvVersionCAS {
			return fmt.Errorf("attempting to add new version vector entry with a CAS that is less than the current version CAS value. Current cas: %s new cas %s", hlv.Version, newVersion.Value)
```

Reply:
I agree, made those changes.